magma/bits.py

```diff
@@ -66,6 +66,17 @@ def __str__(cls):
 class Bits(Array, AbstractBitVector, metaclass=BitsMeta):
     __hash__ = Array.__hash__
 
+    def __init__(self, *args, **kwargs):
```

Comment:
SMT / py support implicit extension and truncations.   I am fine with blocking this behavior and require people to extract or extend.  However, if we are going to replicate in magma then lets be consistent.

Reply:
I think in magma we decided to support implicit extension but not truncation 

CC @phanrahan, what do you think the behavior of calling `Bits[16](a)` when a is of type `Bits[2]` should be? This PR will zext for Bits/UInt and sext for SInt.  Do we want to support truncation as well? E.g. calling `Bits[5](a)` when a is of type `Bits[16]` and just dropping the upper bits?